packages/rpc-transformers/src/tree-traversal.ts

```diff
@@ -1,6 +1,6 @@
 import { RpcRequest, RpcRequestTransformer, RpcResponseTransformer } from '@solana/rpc-spec-types';
 
-export type KeyPathWildcard = { readonly __brand: unique symbol };
+export type KeyPathWildcard = { readonly ['__keyPathWildcard:@solana/kit']: unique symbol };
```

Comment:
I'm pretty sure this one should stay the same, since the `KEYPATH_WILDCARD` depends on the literal identity of the object being the expected one, in order for the tree traversal to identify it. It's being used as a symbol here.

Reply:
If it isn't exported on the package then I think it's fine but if we offer that type to the outside world, we should probably use our new nominal type system.

Reply:
> If it isn't exported on the package then I think it's fine but if we offer that type to the outside world, we should probably use our new nominal type system.

Our new system makes it _worse_. `KEYPATH_WILDCARD` is a token that the tree walker detects via `===`. If you use the `KEYPATH_WILDCARD` from version 2.1.0 with the tree walker from 2.1.1, it _won't detect it as the wildcard_. Keeping this `unique symbol` will make that a type error, rather than confusing runtime behaviour.

The only thing we could do to change this is to make the detection of the keypath wildcard dependent _not_ on the identity of the `{}`, but on the special values of some string (eg. `keyPathWildcard_DO_NO_FORGE_OR_YOU_WILL_BE_FIRED`).